### ISSUE TITLE
The output of `dotnet publish` doesn't work on a machine without .NET Core installed

### ISSUE THREAD

**Issue by christianspecht** [dotnet/cli#813]
*Note: I posted this [as a question on Stack Overflow first](http://stackoverflow.com/q/34580599/6884), because I thought I was doing something wrong.*

My example project is here:
https://github.com/christianspecht/CoreTestVisualStudio

- I created a new project in Visual Studio Community Edition:
*"New Project" ⇒ "Web" ⇒ "Console Application (Package)"*

- I created a new publish profile inside Visual Studio
*("Build" ⇒ "Publish" in the menu)*

Then, I do "Build" ⇒ "Publish" in the menu again, and Visual Studio builds my app.

The result is about 115 MB, (1138 files in 804 folders), and looks like this:

![Visual Studio output](http://i.stack.imgur.com/2ZTXS.png)

When I copy this on another Windows machine without .NET Core installed and execute it, it runs without errors.

---

Now I'm trying to build the same project outside Visual Studio, [via the command line](https://github.com/christianspecht/CoreTestVisualStudio/blob/master/build-cli.bat):

    dotnet publish "%~dp0\src\CoreTestVisualStudio" -c Release -r win7-x64 -o "%~dp0\release\cli"

The result looks different than the first one - only 25 MB / 162 files, all in the same folder:

![dotnet publish output](http://i.stack.imgur.com/PkuMq.png)

Initially the result was even smaller because I forgot to add `Microsoft.NETCore.Runtime` as a dependency, but even after [doing that](https://github.com/christianspecht/CoreTestVisualStudio/commit/ee3d1eaa0b5cd51e8dd1edf9f7717033694c1700), it still does **not** run on my second machine without .NET Core installed:

![error message](https://cloud.githubusercontent.com/assets/263372/12278382/0e21c5f2-b981-11e5-8102-ae3276381fe3.png)

English translation:

> Can't start the program because MSVCP140.dll is missing on this computer. Re-install the program to fix the problem.

Apparently `MSVCP140.dll` [belongs to the C++ runtime](http://stackoverflow.com/q/32998902/6884).

**But:** when I do "Build" ⇒ "Publish" in Visual Studio, the output does **not** contain this DLL, and it works on the other machine anyway.

Maybe I'm still doing something wrong *(and I understand that parts of .NET Core are moving targets at the moment)*, but unfortunately, [the docs about building console apps](http://dotnet.github.io/docs/getting-started/index.html) are not finished yet.

**Comment by piotrpMSFT** [comment]
@christianspecht the CLI is not yet integrated with Visual Studio. Try publishing via the command line interface instead! Please re-open the issue if you see the same behavior from the CLI.

**Comment by christianspecht** [comment]
@piotrpMSFT: unfortunately, I'm not able to re-open the issue myself. But IMO it should be re-opened.

I **did** try to publish via the command line interface.
Here's the short version of the long text in my issue:

0. create project in Visual Studio, publish via Visual Studio
⇒ the result works on a machine without .NET Core installed

0. go to the command line (i.e. outside Visual Studio) and try to publish the same project via [`dotnet publish`](https://github.com/christianspecht/CoreTestVisualStudio/blob/master/build-cli.bat)
⇒ the result does **not** work on a machine without .NET Core installed

---

By the way, it's not related to the fact that I originally created the project in Visual Studio. When I follow the steps in [this tutorial](http://dotnet.github.io/getting-started/) and do `dotnet publish` in the end, the result doesn't work on a machine without .NET Core either.

**Comment by livarcocc** [comment]
Reopening on behalf of @christianspecht

**Comment by piotrpMSFT** [comment]
@christianspecht sorry for closing too fast. I did miss that you dontet-published. This looks like a duplicate of #835 [though that one was created after this one :)] Can you try installing just the C++ Runtime, as you suggested, and validating that this unblocks you? If so, #835 should indeed fix this.

**Comment by xanather** [comment]
Sorry, this may be slightly unrelated, but is there a timeframe on the dotnet CLI visual studio integration, is a alpha/beta release possible? .NET Core w/ CLI seems to work good already, however not being able to debug .NET Core CLI programs in visual studio and trying to make it somewhat work with the old DNX tools is problematic.

**Comment by piotrpMSFT** [comment]
@xanather this is very much on our radar, though I don't think the IDE folks have shared concrete dates yet. We're working on this fairly actively.

@christianspecht if installing the CRT did not fix, can you please reopen?

**Comment by christianspecht** [comment]
@piotrpMSFT Yep, installing the C++ runtime fixed it.
I see that #835 has been resolved in the meantime by [this pull request](https://github.com/dotnet/cli/pull/843). How do I know when this will actually be released?

**Comment by piotrpMSFT** [comment]
Should be in the latest installer. If you're seeing issues, check 'dotnet --version'

**Comment by christianspecht** [comment]
Yes, it works now :+1: 

I downloaded and ran the latest installer, and `dotnet publish`ed again.

Compared to before, I see only two differences in the result of `dotnet publish`:

 - There's one new file (`hostpolicy.dll`)
 - The size of the main .exe increased (418 KB vs 113 KB)

The .exe now works on my second machine without .NET Core installed (of course, I un-installed the C++ runtime before).

**Comment by piotrpMSFT** [comment]
Yea, the size hit is not ideal... must be caused by statically linking CRT.

@schellap, do we have any opportunities for making this smaller?

**Comment by christianspecht** [comment]
Please don't get me wrong, I wasn't complaining. I don't care about the size increase.
It's rather the opposite - I had expected that the size would increase more by including the C++ runtime.
